std/zip.d

```diff
@@ -583,31 +587,17 @@ final class ZipArchive
         if (data.length > uint.max - 2)
             throw new ZipException("zip files bigger than 4 GB are unsupported");
 
-        // Find 'end record index' by searching backwards for signature
-        iend = (data.length > 66_000 ? to!uint(data.length - 66_000) : 0);
-        for (i = to!uint(data.length) - 22; 1; i--)
-        {
-            if (i < iend || i >= data.length)
-                throw new ZipException("no end record");
+        findEndOfCentralDirRecord();
+        i = endrecOffset;
```

Comment:
Move the declaration here then
```suggestion
        uint i = endrecOffset;
```

Some for the following variables.

Reply:
This code seems to be old and there are lot of possibilities for refactoring left (e.g. endcommentlength could be defined later too). In the longrun I'd like to get rid of variable i alltogether by putting everything in separate methods and/or using unions/structs to access the data. For the time being I tried to keep as much of the constructor in the old state and focused on the new findEndOfCentralDirRecord method. But maybe I should do some refactoring first, before I continue to work on issue #20027.